src/main/java/frc/utils/controllers/keyboard/KeyboardController.java

```diff
@@ -0,0 +1,149 @@
+package frc.utils.controllers.keyboard;
+
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.Robot;
+import frc.robot.constants.RobotConstants;
+import frc.utils.applicationsutils.CMDHandler;
+import frc.utils.dashboard.LoggedTableBoolean;
+
+
+/*
+ * inspired from Trigon code
+ *
+ * @author Yoni Kiriaty, Trigon
+ *
+ */
+public class KeyboardController {
+
+    private static final double KEY_PRESSED_VALUE = 0.5;
+
+    private static final String KEYBOARD_TO_NETWORK_TABLES_SIMULATION_CLASS = "keyboard/keyboard_to_nt_simulation.py";
+
+    private static final String KEYBOARD_TABLE = "Keyboard";
+    private static final String KEYS_TAB = "Keys/";
+
+    public final Trigger
+            ESC,
+            F1, F2, F3, F4, F5, F6, F7, F8, F9, F10, F11, F12,
+            DELETE, BACKTICK,
+            ONE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, ZERO,
+            MINUS, EQUALS,
+            BACKSPACE, TAB,
+            Q, W, E, R, T, Y, U, I, O, P, A, S, D, F, G, H, J, K, L, Z, X, C, V, B, N, M,
+            SEMICOLON, APOSTROPHE, LEFT_SHIFT, COMMA, PERIOD, SLASH, RIGHT_SHIFT, LEFT_CONTROL, LEFT_ALT, RIGHT_CONTROL,
+            LEFT_ARROW, RIGHT_ARROW, UP_ARROW, DOWN_ARROW,
+            NUMPAD_0, NUMPAD_1, NUMPAD_2, NUMPAD_3, NUMPAD_4, NUMPAD_5, NUMPAD_6, NUMPAD_7, NUMPAD_8, NUMPAD_9;
+
+    public KeyboardController() {
+        if (Robot.isSimulation() && RobotConstants.ENABLE_KEYBOARD) {
```

Comment:
```suggestion
        if (Robot.isSimulation()) {
```
you check the enable before so no need here, if you init this class it means keyboard is enable